### ISSUE TITLE
Problem with ontology mapping

### ISSUE THREAD

**Issue by nehajain10** [protegeproject/cellfie-plugin#50]
I am working on Rule based ontologies. I have created an ontology and want to add axioms with information about individuals to it. I added the following transformation rule to it:
Rule: 
Individual: @E*(mm:hashEncode)
Types:Mode_Of_Repayment
When I try to create axioms, I get the following error:
Mismatch entity type found in the ontology(Class) and the transformation rule(Individual).
I have this class named Mode_Of_Repayment in my ontology.
Any help regarding solving the error would be welcomed.

**Comment by csnyulas** [comment]
This is more a question for our user mailing list [1](sign up here [2]), not an issue with Cellfie itself.

It would be easiest to debug the issue that you are having, if you could provide your ontology, the (JSON) mapping file, and your spreadsheet. If you can't provide us these, please copy verbatim the error message that you are getting.

FYI: you may nee do add a space after "Types:"

[1] https://mailman.stanford.edu/mailman/listinfo/protege-user
[2] http://protege.stanford.edu/support.php

**Comment by nehajain10** [comment]
Thanks for the prompt reply.
Attaching along the ontology, the JSON mapping file and the spreadsheet.
I have already included a space after "Types:", but the issue still
persists.

**Comment by csnyulas** [comment]
Sorry, but I can't see the attachment in this github issue, neither on the email notification sent by github. Could you please post your question, with the attachments on our mailing lists? Thanks

**Comment by nehajain10** [comment]
Sorry, reattaching the files..
[Loan_Ont.zip](https://github.com/protegeproject/cellfie-plugin/files/235967/Loan_Ont.zip)

**Comment by csnyulas** [comment]
Thank you for the attachments! Without it I could have never figured out what is going on.

Apparently the problem is caused by the fact that the individuals that you try to create from column **B** (such as `Diploma`, `Graduate`, `Doctorate`) and **E** (like `Cash`, `PDC`, `Standing_Instructions`, `Check_Off`) already exist in your ontology as classes. Maybe you have created them in previous attempts to import values from spreadsheets. If you have a clean ontology, that does not have these classes, which are suppose to be imported from the spreadsheet as instances, your rules should work. In fact, when I removed the rule referring to column **E** and I changed the range of the rule referring to column **B** to go only from row 6 to row 7 (containing values that are not class names in your ontology), I did not get the exception.

Basically, it looks like Cellfie/MappingMaster does not allow you to pun classes as individuals (or anything else, I suppose), nor to use classes as individuals, even if they have been already punned as individual.
